rust/src/wrapper/assigned_numbers/mod.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Assigned numbers from the Bluetooth spec.
+
+use crate::wrapper::core::Uuid16;
+use lazy_static::lazy_static;
+use pyo3::{
+    intern,
+    types::{PyDict, PyModule},
+    PyResult, Python,
+};
+use std::collections;
+
+mod services;
+
+pub use services::SERVICE_IDS;
+
+lazy_static! {
+    /// Assigned company IDs
+    pub static ref COMPANY_IDS: collections::HashMap<Uuid16, String> = load_company_ids()
+    .expect("Could not load company ids -- are Bumble's Python sources available?");
+
+}
+
+fn load_company_ids() -> PyResult<collections::HashMap<Uuid16, String>> {
```

Comment:
If we had the guarantee that `company_ids.py` always stuck to a known simple layout with one entry per lines, would it make sense to generate `company_ids.rs` from it at build time with a trivial parser?

Reply:
I think if parsing is involved, at that point it might make more sense to just have one csv/json/yaml/etc file that both could load at startup / compile time as appropriate.

Reply:
Wouldn't it be best to do the parsing at build/compile time? For rust, definitely faster. For Python, moderately faster (except maybe when the python is first compiled to a .pyc).

Reply:
What we could do is spin up the python VM in build.rs to do this load step (no manual parsing needed), then emit a .rs file 🤔